src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,112 @@
+        /// middleware to.</param>
+        /// <returns>A convention routes for the Prometheus scraping endpoint.</returns>
+        public static IEndpointConventionBuilder MapPrometheusScrapingEndpoint(this IEndpointRouteBuilder endpoints)
+            => MapPrometheusScrapingEndpoint(endpoints, path: null, meterProvider: null, configureBranchedPipeline: null);
+
+        /// <summary>
+        /// Adds OpenTelemetry Prometheus scraping endpoint middleware to an
+        /// <see cref="IEndpointRouteBuilder"/> instance.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add
+        /// middleware to.</param>
+        /// <param name="path">The path to use for the branched pipeline.</param>
+        /// <returns>A convention routes for the Prometheus scraping endpoint.</returns>
+        public static IEndpointConventionBuilder MapPrometheusScrapingEndpoint(this IEndpointRouteBuilder endpoints, string path)
+        {
+            Guard.ThrowIfNull(path);
+            return MapPrometheusScrapingEndpoint(endpoints, path, meterProvider: null, configureBranchedPipeline: null);
+        }
+
+        /// <summary>
+        /// Adds OpenTelemetry Prometheus scraping endpoint middleware to an
+        /// <see cref="IEndpointRouteBuilder"/> instance.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add
+        /// middleware to.</param>
+        /// <param name="path">Optional path to use for the branched pipeline.
+        /// If not provided then <see cref="PrometheusExporterOptions.ScrapeEndpointPath"/>
+        /// is used.</param>
+        /// <param name="meterProvider">Optional <see cref="MeterProvider"/>
+        /// containing a <see cref="PrometheusExporter"/> otherwise the primary
+        /// SDK provider will be resolved using application services.</param>
+        /// <param name="configureBranchedPipeline">Optional callback to
+        /// configure the branched pipeline. Called before registration of the
+        /// Prometheus middleware.</param>
+        /// <returns>A convention routes for the Prometheus scraping endpoint.</returns>
+        public static IEndpointConventionBuilder MapPrometheusScrapingEndpoint(
+            this IEndpointRouteBuilder endpoints,
+            string path = null,
+            MeterProvider meterProvider = null,
+            Action<IApplicationBuilder> configureBranchedPipeline = null)
+        {
+            var builder = endpoints.CreateApplicationBuilder();
+
+            // Note: Order is important here. MeterProvider is accessed before
+            // GetOptions<PrometheusExporterOptions> so that any changes made to
+            // PrometheusExporterOptions in deferred AddPrometheusExporter
+            // configure actions are reflected.
+            meterProvider ??= builder.ApplicationServices.GetRequiredService<MeterProvider>();
```

Comment:
Using the [ASP.NET Core health checks extensions](https://github.com/dotnet/aspnetcore/blob/v6.0.5/src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs) as an example, I see they use `endpoints.ServiceProvider` instead of `builder.ApplicationServices` when resolving things. I'm guessing behind the scenes it evaluates to the same thing, but it'd probably be better to use `endpoints.ServiceProvider` for resolving here to be consistent with that.

Reply:
I've quickly made that small change.

Before running the GitHub workflows again, there was an error in the [previous run](https://github.com/open-telemetry/opentelemetry-dotnet/runs/6596758825) that I need to fix around API compatibility. I'm unsure what I need to do here. Are you able to tell me what I need to do here?

Reply:
Look in the [.publicApi](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.Prometheus/.publicApi) folder. There are text files with the shipped and unshipped public methods listed. Yours is unshipped. You need to add these to the list.

Might also be good to add this to the CHANGELOG.md file.

There's more info in the CONTRIBUTING.md.

Reply:
Cool I've now made changes to `.publicApi/netcoreapp3.1/PublicAPI.Unshipped.txt` and CHANGELOG.md file.